### ISSUE TITLE
Widgets screen: Can't tell which Legacy Widgets are on my screen without selecting them

### ISSUE THREAD

**Issue by noisysocks** [WordPress/gutenberg#24566]
When I first open the new Widgets screen, I'm greeted with this:

<img width="713" alt="Screen Shot 2020-08-14 at 15 16 07" src="https://user-images.githubusercontent.com/612155/90216168-31ebd100-de41-11ea-8d90-e694c8596f64.png">

Because we're hiding the title from the Legacy Widget block until it's selected, it's really hard to tell what widgets I have in my widget areas.

How about we **always** show the title? 

<img width="714" alt="Screen Shot 2020-08-14 at 15 13 03" src="https://user-images.githubusercontent.com/612155/90216216-4a5beb80-de41-11ea-97ef-52daa9b0b6e7.png">

**Comment by ZebulanStanphill** [comment]
Since the legacy widgets here aren't acting as a front-end preview, I think we should definitely give them visible borders and text labels. I guess in general they should look more like the placeholders on various blocks (e.g. Image, Video, etc.), with solid backgrounds and a clear visual look that indicates "I'm a set of plain controls that isn't WYSIWYG at all".

**Comment by afercia** [comment]
Also, as reported in another issue, all the widgets should be listed with a H3 heading in the same way it happens now in the legacy Widgets screen.

**Comment by mapk** [comment]
Good points here. @noisysocks, I agree when there's tons of forms from various widgets, it's important to communicate which widget they apply to and group them somehow so they don't bleed into other widget forms visually.

I see this working is like this:

**Unselected (resting) - requiring input to work**
When a Legacy widget block is unselected (resting) but requires necessary form inputs to work, it should show the "Edit" view.

![unselected requiring input](https://user-images.githubusercontent.com/617986/92636742-b983fe80-f28c-11ea-8075-800bce6f4998.png)

**Unselected (resting) - not requiring input to work**
When the Legacy widget block is unselected and does not require any input to show properly, it should default to the "Preview" view.

![unselected not requiring input](https://user-images.githubusercontent.com/617986/92636856-e9cb9d00-f28c-11ea-859c-8fec4c0c1bcf.png)

**Selected (focused)**
Any Legacy widget block that has focus automatically defaults to the "Edit" view unless changed manually to the "Preview" view, but if unselected and selected again, it defaults again to the "Edit" view.

![selected](https://user-images.githubusercontent.com/617986/92636921-036ce480-f28d-11ea-9205-4651e273014c.png)

**Comment by ZebulanStanphill** [comment]
@mapk Nice mockups. That looks a lot better than the current situation.

**Comment by afercia** [comment]
> I agree when there's tons of forms from various widgets, it's important to communicate which widget they apply to and group them somehow so they don't bleed into other widget forms visually.

I'd totally second this. And it needs to be communicated also _semantically_, not only visually. So that assistive technologies can understand the grouping and communicate it to users.

**Comment by mapk** [comment]
Looks like we've got some good direction here. 🎉  Thanks, everyone!

**Comment by adamziel** [comment]
Merging https://github.com/WordPress/gutenberg/pull/25638 marked a first step towards this. For now, all legacy widgets default to forms with a visible title bar. The next step would be detecting which widgets require user input to generate preview, and defaulting to preview for these.

**Comment by afercia** [comment]
Glad to see the progress made on #25638. Question: why the widget titles aren't H3 headings as asked for in this issue and in #24561? A good headings hierarchy is fundamental not only for better semantics but also as a navigational tool for assistive technology users.

**Comment by adamziel** [comment]
Good point @afercia - let's definitely make them H3 headings

**Comment by kevin940726** [comment]
Accidentally closed this via #25690, reopen it now.
